### PR TITLE
Bugfix for ssl_versions in rabbitmq.config 

### PR DESCRIPTION
**Pull request by dalees**
RabbitMQ was not respecting this setting. The format of the configuration file was incorrect. This patch updates to bring this in line with documentation.

  The format reference example is given in https://www.rabbitmq.com/ssl.html#disabling-tls-versions
  Also add version list for rabbitmq_management config ssl_opts.

### CODE REVIEW

File: templates/rabbitmq.config.erb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 % This file managed by Puppet
 % Template Path: <%= @module_name %>/templates/rabbitmq.config
 [
+<%- if @ssl and @ssl_versions -%>
+  {ssl, [{versions, [<%= @ssl_versions.sort.map { |v| "'#{v}'" }.join(', ') %>]}]},
+<%- end -%>
```
Comment (jtopjian) [inline templates/rabbitmq.config.erb]: Does the placement of this block make a difference with the behavior of rabbit? My original pull request had this block placed here as well, but we moved it further down in order to combine conditional statements.

File: templates/rabbitmq.config.erb [hunk 2]
```diff
@@ -16,18 +19,16 @@
...
-                    <%- if @ssl_versions -%>
-                      ,{ssl, [{versions, [<%= @ssl_versions.sort.map { |v| "'#{v}'" }.join(', ') %>]}]}
-                    <% end -%>]},
+    {ssl_options, [<%- if @ssl_cacert != 'UNSET' -%>{cacertfile,"<%= @ssl_cacert %>"},
+                   <% end %>{certfile,"<%= @ssl_cert %>"},
+                   {keyfile,"<%= @ssl_key %>"},
+                   {verify,<%= @ssl_verify %>},
+                   {fail_if_no_peer_cert,<%= @ssl_fail_if_no_peer_cert %>}
+                   <%- if @ssl_versions -%>
+                   ,{versions, [<%= @ssl_versions.sort.map { |v| "'#{v}'" }.join(', ') %>]}
```
Comment (jtopjian) [inline templates/rabbitmq.config.erb]: I really have no idea how I missed this multiple times. Thanks for catching it.

File: templates/rabbitmq.config.erb (hunk 1 above)
Comment (dalees) [inline templates/rabbitmq.config.erb]: No, because it doesn't actually do anything functional in my tests.
The docs place it outside the rabbit config scope, so that is where it should be unless the docs are wrong.

**Comment by jtopjian** [comment]
Looks like you'll need to update the rspec tests as well. Line 521 of `spec/classes/rabbitmq_spec.rb` should be the only modification needed.

**Comment by dalees** [comment]
Just saw the rspec failures, thanks. Will get these updated as soon as I get these running myself.

File: templates/rabbitmq.config.erb (hunk 1 above)
Comment (jtopjian) [inline templates/rabbitmq.config.erb]: I agree with this.

File: templates/rabbitmq.config.erb [hunk 3]
```diff
@@ -16,18 +19,16 @@
...
-    {ssl_options, [<%- if @ssl_cacert != 'UNSET' -%>{cacertfile,"<%= @ssl_cacert %>"},<%- end -%>
-                    {certfile,"<%= @ssl_cert %>"},
-                    {keyfile,"<%= @ssl_key %>"},
-                    {verify,<%= @ssl_verify %>},
-                    {fail_if_no_peer_cert,<%= @ssl_fail_if_no_peer_cert %>}
-                    <%- if @ssl_versions -%>
-                      ,{ssl, [{versions, [<%= @ssl_versions.sort.map { |v| "'#{v}'" }.join(', ') %>]}]}
-                    <% end -%>]},
+    {ssl_options, [<%- if @ssl_cacert != 'UNSET' -%>{cacertfile,"<%= @ssl_cacert %>"},
+                   <% end %>{certfile,"<%= @ssl_cert %>"},
```
Comment (cmurphy) [inline templates/rabbitmq.config.erb]: Why did you move the `<%- end -%>` here?

Comment (dalees) [inline templates/rabbitmq.config.erb]: Only for prettiness in the output config when no ssl_cacert is specified, though it sacrifices some readability in the template code.
Which should we prefer? I've seen both in this template.

Comment (cmurphy) [inline templates/rabbitmq.config.erb]: We could keep it where it was but change `<%- end -%>` to `<% end %>`, then the template looks okay and the output is prettier.

Comment (dalees) [inline templates/rabbitmq.config.erb]: On discussion with my colleague we decided these whitespace changes shouldn't be in this pull request at all. I've got some more changes to come later so I'll add these separately.

**Comment by cmurphy** [comment]
Thanks @dalees and @jtopjian!
